pycc/lccwfn.py

```diff
@@ -0,0 +1,476 @@
+        Returns
+        -------
+        elcc: float 
+            lCC correlation energy 
+        """
+        lcc_tstart = time.time()
+
+        #initialize variables for timing each function
+        #self.fae_t = Timer("Fae")
+        #self.fme_t = Timer("Fme")
+        #self.fmi_t = Timer("Fmi")
+        #self.wmnij_t = Timer("Wmnij")
+        #self.zmbij_t = Timer("Zmbij")
+        #self.wmbej_t = Timer("Wmbej")
+        #self.wmbje_t = Timer("Wmbje")
+        #self.tau_t = Timer("tau")
+        #self.r1_t = Timer("r1")
+        #self.r2_t = Timer("r2")
+        #self.energy_t = Timer("energy")
+
+        #ldiis = helper_ldiis(self.t1_ii, self.t2_ij, max_diis) 
+        
+        elcc = self.lcc_energy(self.Local.Fov_ij,self.Local.Loovv_ij,self.t1_ii, self.t2_ij)
+        print("CC Iter %3d: lCC Ecorr = %.15f dE = % .5E MP2" % (0,elcc,-elcc))
+
+        for niter in range(1, maxiter+1):
+
+            elcc_last = elcc
+
+            r1_ii, r2_ij = self.local_residuals(self.t1_ii, self.t2_ij)
+
+            rms = 0
+            rms_t1 = 0
+            rms_t2 = 0
+
+            for i in range(self.no):
+                for j in range(self.no):
+                    ij = i*self.no + j
+
+                    self.t2_ij[ij] -= r2_ij[ij]/(self.eps[ij].reshape(1,-1) + self.eps[ij].reshape(-1,1)
+                    - self.H.F[i,i] - self.H.F[j,j])
+
+                    rms_t2 += contract('ZY,ZY->',r2_ij[ij],r2_ij[ij])
+
+            rms = np.sqrt(rms_t2)
+            elcc = self.lcc_energy(self.Local.Fov_ij,self.Local.Loovv_ij,self.t1_ii, self.t2_ij)
+            ediff = elcc - elcc_last
+            print("lCC Iter %3d: lCC Ecorr = %.15f  dE = % .5E  rms = % .5E" % (niter, elcc, ediff, rms))
+
+            # check for convergence
+            if ((abs(ediff) < e_conv) and rms < r_conv):
+                print("\nlCC has converged in %.3f seconds.\n" % (time.time() - lcc_tstart))
+                print("E(REF)  = %20.15f" % self.eref)
+                print("E(%s) = %20.15f" % (self.local + "-" + self.model, elcc))
+                print("E(TOT)  = %20.15f" % (elcc + self.eref))
+                self.elcc = elcc
+                #print(Timer.timers)
+                return elcc
+
+            #ldiis.add_error_vector(self.t1_ii,self.t2_ij)
+            #if niter >= start_diis:
+                #self.t1_ii, self.t2_ij = ldiis.extrapolate(self.t1_ii, self.t2_ij)
+
+    def local_residuals(self, t1_ii, t2_ij):
+        """
+        Constructing the two- and four-index intermediates
+        Then evaluating the singles and doubles residuals, storing them in a list of length occ (single) and length occ*occ (doubles)
+        Naming scheme same as integrals where _ij is attach as a suffix to the intermediate name 
+        ... special naming scheme for those involving two different pair space ij and im -> _ijm
+
+        To do
+        ------
+        There are some arguments that isn't really being used and will be updated once things are good to go
```

Comment:
I don't understand what's going on here.  If the arguments aren't needed, why not remove them?

Also: isn't --> aren't